[JIT] Remove references to no longer generated _tanh_backward and _sigmoid_backward

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44138 [JIT] Remove references to no longer generated _tanh_backward and _sigmoid_backward**
* #44137 [TensorExpr fuser] Guard nodes that have tensor output properties determined by non-tensor inputs

If you look at the sigmoid and tanh backward they are composed of other ops: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L786
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L164

So tanh_backward and sigmoid_backward are no longer generated / legacy ops.

Differential Revision: [D23543603](https://our.internmc.facebook.com/intern/diff/D23543603)

Commit:
Remove no longer generated _tanh_backward and _sigmoid_backward

[ghstack-poisoned]

Commit:
Update on "[JIT] Remove references to no longer generated _tanh_backward and _sigmoid_backward"


If you look at the sigmoid and tanh backward they are composed of other ops: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L786
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L164

So tanh_backward and sigmoid_backward are no longer generated / legacy ops. 


[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/codegen/fuser/codegen.cpp:
```diff
@@ -254,8 +254,6 @@ static std::string encodeRHS(const Node* n) {
       {aten::where, "(${0} ? ${1} : ${2})"},
 
       // simple derivatives
```
Probably want to delete this comment too, right?

bertmaher reviewed (approved):
nice cleanup!

Commit:
Update on "[JIT] Remove references to no longer generated _tanh_backward and _sigmoid_backward"


If you look at the sigmoid and tanh backward they are composed of other ops: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L786
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L164

So tanh_backward and sigmoid_backward are no longer generated / legacy ops. 


[ghstack-poisoned]

Commit:
Update on "[JIT] Remove references to no longer generated _tanh_backward and _sigmoid_backward"


If you look at the sigmoid and tanh backward they are composed of other ops: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L786
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_script.cpp#L164

So tanh_backward and sigmoid_backward are no longer generated / legacy ops. 


[ghstack-poisoned]
